cmd/config.go

```diff
@@ -0,0 +1,91 @@
+package cmd
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	electricityMapAPITokenEnvVar = "ELECTRICITY_MAP_API_TOKEN"
+	wattTimeUserEnvVar           = "WATT_TIME_USER"
+	wattTimePasswordEnvVar       = "WATT_TIME_PASSWORD"
+)
+
+func getConfigFile() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", nil
+	}
+
+	return filepath.Join(homeDir, configDir, configFileName), nil
+}
+
+func configFileExists(configFile string) (bool, error) {
+	_, err := os.Stat(configFile)
+	if err == nil {
+		return true, nil
+	}
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	}
+
+	return false, err
+}
+
+func readConfig() (string, string, error) {
+	configFile, err := getConfigFile()
+	if err != nil {
+		return "", "", err
+	}
+
+	viper.SetConfigFile(configFile)
+
+	err = viper.ReadInConfig()
+	if errors.Is(err, os.ErrNotExist) {
+		// Config file may not be available e.g. when running as a container.
```

Comment:
The config file support and docker where not playing nicely so I refactored this part.

We now try to use the config file but it can't be written we still use the CLI arguments or env vars. As this will work better when run in nomad or k8s.

Reply:
oh thanks Ross.

this is what the `providerName := viper.GetString(provider)` gubbins is all about, right?